specification/web/resource-manager/Microsoft.Web/stable/2023-12-01/CommonDefinitions.json

```diff
@@ -3205,7 +3206,8 @@
           },
           "x-ms-identifiers": [
             "name"
-          ]
+          ],
+          "x-ms-secret" : true
         },
         "metadata": {
```

Comment:
for metadata and settings , you can also model this as a separate property like protectedmetadata and protectedsettings , since this way it will cause a breaking change. 

see protectedsettings vs settings in vm for an example : 

[https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/virtualmachines/extensions?pivots=deployment-language-arm-template#virtualmachineextensionproperties-1](https://learn.microsoft.com/en-us/azure/templates/microsoft.compute/virtualmachines/extensions?pivots=deployment-language-arm-template#virtualmachineextensionproperties-1)